extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/AuthenticationRedirectException.java

```diff
@@ -19,8 +19,12 @@ public AuthenticationRedirectException(int code, String redirectUri) {
         this.redirectUri = redirectUri;
     }
 
+    public AuthenticationRedirectException(Boolean autoRedirect, String redirectUri) {
```

Comment:
IMO, this logic belongs to the `CodeAuthenticationMechanism` as it is really related to this flow. See my other comment around using `401`and `403`.

Reply:
That makes sense. I will remove this and put the logic in CodeAuthenticationMechanism using the exception constructor that allows for the specification of the return code.